### PR TITLE
Prevent Y log scale with negative data

### PR DESCRIPTION
**Pull request by Moo-Marc**
Avoid errors when trying to apply y axis log scale on log-power data.  

Strangely, the y scale log option in the "display configuration" button on psd figures activates the function SetScaleModeY in figure_timeseries, while in the right-click menu, it activates the function ToggleLogScaleY in figure_spectrum.  That confused me for a while.

### CODE REVIEW

**Comment by Moo-Marc** [comment]
I think I ironed out most kinks.  I only tested with PSD plots.  I'm not sure if these changes would affect time-freq plots?

File: toolbox/gui/figure_timeseries.m [hunk 1]
```diff
@@ -4074,13 +4074,22 @@ function SetScaleModeX(hFig, newMode)
 
 %% ===== SET Y-SCALE MODE =====
 function SetScaleModeY(hFig, newMode)
+    [hFig, iFig, iDS] = bst_figures('GetFigure', hFig);
```
Comment (ftadel) [inline toolbox/gui/figure_timeseries.m:4077]: Use the accessor `Handles = bst_figures('GetFigureHandles', hFig);` instead

**Review by ftadel — CHANGES_REQUESTED** [review]

**Comment by ftadel** [comment]
> the y scale log option in the "display configuration" button on psd figures activates the function SetScaleModeY in figure_timeseries, while in the right-click menu, it activates the function ToggleLogScaleY in figure_spectrum

Indeed, this is not optimal to have two ways of handling this. Maybe replacing the Xlog and Ylog checkboxes for the display menu of the spectrum figures with two radio buttons (Linear/Log) as in the popup menu would be easier to read. And it would allow to call only one function (SetScaleModeY) and remove the other one (ToggleLogScaleY).
If the handling of X and Y axis is similar in terms of code, it would be better.
Feel free to propose something simpler. 

> I'm not sure if these changes would affect time-freq plots?

There are no reference to these functions in figure_timefreq, so it won't affect the TF plots.
However, the timefreq files (Hilbert, Morlet, coherence, spectral Granger, PLV, ...) can be displayed as spectrum figures, you should check that all this still works as expected.

File: toolbox/gui/figure_spectrum.m [hunk 2]
```diff
@@ -1222,6 +1234,19 @@ function UpdateFigurePlot(hFig, isForced)
     end
         
     % ===== DISPLAY =====
+    % Get preferred scale modes.
+    if ~isfield(TsInfo, 'XScale')
```
[the remark this answers is not in the document]
Comment (Moo-Marc) [reply toolbox/gui/figure_spectrum.m:1238]: These tests were already there, just a little lower.  I just tested and apparently they're not needed, so agreed.

**Comment by Moo-Marc** [comment]
> And it would allow to call only one function (SetScaleModeY) and remove the other one (ToggleLogScaleY)

It was simple to just reuse SetScaleModeY in both cases, but I'll keep the gui as is for now.  I personally prefer the checkmark in the display menu.  The pop-up menu "Y scale: log" is a bit ambiguous as to whether it shows the current setting or if it means "Set Y scale to: log".  But that menu has icons where the checkmark would go.

**Comment by Moo-Marc** [comment]
Tested TF-linked spectrum figure.  Works fine.
There's still an issue with FOOOF spectra (probably to do with their NaNs), but there will be another PR for that and other fooofy stuff soon.

**Comment by Moo-Marc** [comment]
Last commit before updating the tutorial screenshots. 
Display the real units in spectrum, e.g. fT for MEG.
Plus fixed a bug on TF-linked timeseries figure (auto scale y didn't work).

**Comment by Moo-Marc** [comment]
Updated tutorial screenshots and What's New.

**Pull request merged**